widgyts/dataset_viewer.py

```diff
@@ -199,9 +200,30 @@ def widget(self):
 
         traitlets.link((dropdown, "value"), (self, "grid_colormap"))
 
+        button = ipywidgets.Button(description="Positions")
+
+        positions_list = []
+
+        def on_button_clicked(b):
+            positions_list = positions_list + traitlets.List(
+                list(self.renderer.camera.position)
+            )
+            print(positions_list)
+
+        button.on_click(on_button_clicked)
+
+        select = ipywidgets.Select(
+            options=positions_list,
```

Comment:
`options = self.positions_list,`

but, keep in mind, this will just be the initialization.  We need to `observe` the `position_list` for changes, and then update `select.options` accordingly.

One other thing to note is that the `position_list` will be a list of 3-element lists or arrays.  What we want is to use the tuple syntax, which provides a "display name" and a "value" to `ipywidgets.Select`.  So for instance, this could be something like:

`("Value One", 1), ("Value Two", 2) ...`

For our purposes, you could accomplish this with something along the lines of `[ ("Position %s" % i, v) for i, v in enumerate(self.position_list)]` .

Reply:
More info: https://ipywidgets.readthedocs.io/en/latest/examples/Widget%20List.html#Selection-widgets